lib.php

```diff
@@ -0,0 +1,50 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Mail log plugin by Catalyst IT.
+ * Update Moodle user bounce counts and delete mail message
+ * Delete old bounce messages
+ * Expunge
+ *
+ * @package local_maillog
+ * @author  Sumaiya Javed <sumaiya.javed@catalyst.net.nz>
+ * @license http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+defined('MOODLE_INTERNAL') || die();
+
+
+/**
+ * Records the date of the last update to queuemails admin setting.
+ *
+ * @return void
+ */
+function local_maillog_notify() {
+    // Add time when the queuemails setting was updated.
+    global $DB;
+    $name = 'queuemailsdate'; $plugin = 'local_maillog'; $now = time();
+    $id = $DB->get_field('config_plugins', 'id', array('name' => $name, 'plugin' => $plugin));
```

Comment:
is there a reason this isn't using the get_config() api ? - could end up with weird behaviour as it's also cached.

Reply:
I see a single set_config() is doing the job. Thanks for your feedback.